lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -46,8 +60,8 @@ defmodule Kernel.LexicalTracker do
 
   # Starts the tracker and returns its pid.
   @doc false
-  def start_link(dest) do
-    :gen_server.start_link(__MODULE__, dest, [])
+  def start_link(dest, parent) do
```

Comment:
@antipax I may be missing something but I don't understand why this is needed.

> Before now, all modules in a file would actually end up using the same tracker, which resulted in all modules after the first containing all of the references of the ones defined above them. 

The warnings you are calculating are per *file*. In fact, this change probably means we would never emit a warning for unused functions at the end of a file (outside of a module) like this:

```elixir
defmodule Foo do
  # ...  
end
Enum.map([], fn _ -> IO.i_dont_exist end)
```

If there is an issue in this code, I would rather say the issue is in the Mix compiler which is tracking all this data per module while it should be doing so per file. We will need to change how we store data in the manifest. Please ping me later when you have some time. :)